tensorflow_datasets/translate/wmt.py

```diff
@@ -731,7 +731,7 @@ def _get_local_paths(ds, extract_dirs):
         extract_dirs = extract_dirs * len(rel_paths)
       return [os.path.join(ex_dir, rel_path) if rel_path else ex_dir
               for ex_dir, rel_path in zip(extract_dirs, rel_paths)]
-
+    idx = 0
     for ss_name in split_subsets:
```

Comment:
The S3 key needs to be deterministically generated per example, here because `split_subsets` is generated from a dict, I do not thing the order is constant and deterministically generated.
Additionally it there is a two level loop, so we need to take special attention to the key. 

My suggestion is to have each subgenerator (`_parse_tvs`, ...) yield an id key and then combine with the split_subsets key:

```
for ss_name in split_subsets:
  ...
  for sub_key, ex in sub_generator(*files):
    ...
    key = '{}/{}'.format(ss_name, sub_key)
    yield key, ex
```

Then you need to be careful that each sub-generator generate a deterministic key.

Reply:
Okay!